src/debugger/gdb-stub.c

```diff
@@ -210,6 +211,62 @@ static void _step(struct GDBStub* stub, const char* message) {
 	UNUSED(message);
 }
 
+static void _writeMemoryBinary(struct GDBStub* stub, const char* message) {
+	const char* readAddress = message;
+	unsigned i = 0;
+	uint32_t address = _readHex(readAddress, &i);
+	readAddress += i + 1;
+
+	i = 0;
+	uint32_t size = _readHex(readAddress, &i);
+	readAddress += i + 1;
+
+	if (size > 512) {
+		_error(stub, GDB_BAD_ARGUMENTS);
+		return;
+	}
+
+	struct ARMCore* cpu = stub->d.core->cpu;
+	for (i = 0; i < size; i++) {
+		// Parse escape char
+		uint8_t byte;
```

Comment:
Please do the assignment here and don't increment the variable in the middle of an expression, for readability.